### PR TITLE
Adding ScrollViewer anchor registering/unregistering public methods

### PR DESCRIPTION
**Pull request by RBrid**
Issue #264 

The ScrollViewer is now exposing two new RegisterAnchorCandidate/UnregisterAnchorCandidate methods to explicitly register and unregister anchor candidates. The calls are being forwarded to the inner Scroller's IAnchorProvider implementation (with the same method signatures RegisterAnchorCandidate & UnregisterAnchorCandidate).

Added new VerifyAnchorCandidateRegistration test to exercise the two new methods.

### CODE REVIEW

File: dev/ScrollViewer/ScrollViewer.cpp [hunk 1]
```diff
@@ -162,6 +162,42 @@ void ScrollViewer::InputKind(winrt::InputKind const& value)
...
 
+void ScrollViewer::RegisterAnchorCandidate(winrt::UIElement const& element)
+{
+    SCROLLVIEWER_TRACE_VERBOSE(*this, TRACE_MSG_METH_PTR, METH_NAME, this, element);
+
+    if (auto scroller = m_scroller.get())
+    {
+        const winrt::Controls::IScrollAnchorProvider scrollerAsAnchorProvider = scroller.try_as<winrt::Controls::IScrollAnchorProvider>();
+
+        if (scrollerAsAnchorProvider)
```
Comment (ranjeshj) [inline dev/ScrollViewer/ScrollViewer.cpp]: nit: can simplify to 
if(const auto scrollAnchorProvider  =...)
{
}

File: dev/ScrollViewer/ScrollViewer.idl [hunk 2]
```diff
@@ -79,6 +79,8 @@ unsealed runtimeclass ScrollViewer : Windows.UI.Xaml.Controls.Control
     [MUX_DEFAULT_VALUE("ScrollViewer::s_defaultAnchorRatio")]
     [MUX_PROPERTY_VALIDATION_CALLBACK("ValidateAnchorRatio")]
     Double VerticalAnchorRatio { get; set; };
+    void RegisterAnchorCandidate(Windows.UI.Xaml.UIElement element);
```
Comment (ranjeshj) [inline dev/ScrollViewer/ScrollViewer.idl]: what is the scenario ? I thought the anchor requested event was the way to achieve this.

File: dev/ScrollViewer/ScrollViewer.cpp (hunk 1 above)
Comment (RBrid) [reply dev/ScrollViewer/ScrollViewer.cpp]: Done. Thanks.

File: dev/ScrollViewer/ScrollViewer.idl (hunk 2 above)
Comment (RBrid) [reply dev/ScrollViewer/ScrollViewer.idl]: The idea was to expose the same methods at the ScrollViewer level as the Scroller level (without implementing the interface in both places though), as a convenient pass through to the inner Scroller control. It is true that handling the AnchorRequested event lets you populate the candidates too. Let's check it in and confirm with Micah when he comes back.

**Pull request merged**